source/persistency/PetaloPersistencyManager.cc

```diff
@@ -389,6 +392,41 @@ void PetaloPersistencyManager::StoreSensorHits(G4VHitsCollection* hc)
   }
 }
 
+
+void PetaloPersistencyManager::StoreChargeHits(G4VHitsCollection* hc)
+{
+  ChargeHitsCollection* hits = dynamic_cast<ChargeHitsCollection*>(hc);
+  if (!hits) return;
+
+  std::vector<G4int > sensor_ids;
+  for (size_t i=0; i<hits->entries(); i++) {
+
+    ChargeHit* hit = dynamic_cast<ChargeHit*>(hits->GetHit(i));
+    if (!hit) continue;
+
+    wire_bin_size_ = hit->GetBinSize();
+    const std::map<G4double, G4int>& wvfm = hit->GetChargeWaveform();
+
+    std::map<G4double, G4int>::const_iterator it;
+    for (it = wvfm.begin(); it != wvfm.end(); ++it) {
+      unsigned int time_bin = (unsigned int)((*it).first/wire_bin_size_+0.5);
+      unsigned int charge   = (unsigned int)((*it).second+0.5);
```

Comment:
Can you remind me why do we need to add `0.5` in `time_bin` and `charge`? Is it related to the bin_width?

Reply:
It's a way to apply the rounding correctly. In fact, it's probably obsolete, as explained here https://stackoverflow.com/questions/47302544/why-do-lots-of-old-programs-use-floor0-5-input-instead-of-roundinput, for instance. Anyway, before changing it, I would a sanity check and not in this PR.